flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -1134,4 +1141,28 @@ private Configuration getOperatorRestConfig(Configuration origConfig) throws IOE
                         });
         return conf;
     }
+
+    protected void scaleJmToZero(
+            EditReplacePatchable<Deployment> jmDeployment, String namespace, String clusterId) {
+        LOG.info("Scaling down JM deployment to 0 before deletion");
```

Comment:
Since we also have a possible timeout here, maybe it's worth adding the duration of the timeout in the log